### PR TITLE
Fix test for Substrate#9898 

### PR DESCRIPTION
**Pull request by emostov**
target branch: kiz-improve-ocw-resubmit
target pr: https://github.com/paritytech/substrate/pull/9898

Prior to #9898, the election OCW cleared its cached solution whenever it saw events from `do_elect`. Now, the election OCW only clears its solution when run on the first block of the unsigned phase. In this PR the relevant test is updated to try and document the change.

### CODE REVIEW

File: frame/election-provider-multi-phase/src/unsigned.rs [hunk 1]
```diff
@@ -1241,35 +1241,62 @@ mod tests {
...
 	#[test]
-	fn ocw_clears_cache_after_election() {
-		let (mut ext, _pool) = ExtBuilder::default().build_offchainify(0);
+	fn ocw_clears_cache_on_unsigned_phase_open() {
+		let (mut ext, pool) = ExtBuilder::default().build_offchainify(0);
 		ext.execute_with(|| {
-			roll_to(25);
-			assert_eq!(MultiPhase::current_phase(), Phase::Unsigned((true, 25)));
+			const BLOCK: u64 = 25;
+			let block_plus = |delta: u64| BLOCK + delta;
```
Comment (emostov) [inline frame/election-provider-multi-phase/src/unsigned.rs:1248]: this is an idiom taken from the other tests in this file

File: frame/election-provider-multi-phase/src/unsigned.rs [hunk 2]
```diff
@@ -1241,35 +1241,62 @@ mod tests {
...
 
-			// after an election, the solution must be cleared
+			// record the submitted tx,
+			let tx_cache_1 = pool.read().transactions[0].clone();
+			// and assume it has been processed.
+			pool.try_write().unwrap().transactions.clear();
+
+			// after an election, the solution is not cleared
 			// we don't actually care about the result of the election
-			roll_to(26);
```
Comment (emostov) [inline frame/election-provider-multi-phase/src/unsigned.rs:1269]: afaict every call to `roll_to` after  the first is not relevant to the test

File: frame/election-provider-multi-phase/src/unsigned.rs [hunk 3]
```diff
@@ -1241,35 +1241,62 @@ mod tests {
...
-			roll_to(25);
-			assert_eq!(MultiPhase::current_phase(), Phase::Unsigned((true, 25)));
+			const BLOCK: u64 = 25;
+			let block_plus = |delta: u64| BLOCK + delta;
+			let offchain_repeat = <Runtime as Config>::OffchainRepeat::get();
 
-			// we must clear the offchain storage to ensure the offchain execution check doesn't get
-			// in the way.
-			let mut storage = StorageValueRef::persistent(&OFFCHAIN_LAST_BLOCK);
-			storage.clear();
```
[the remark this answers is not in the document]
Comment (kianenigma) [reply frame/election-provider-multi-phase/src/unsigned.rs:1253]: I actually always thought it to be an overkill and wanted to remove it :p

**Review by kianenigma — APPROVED** [review]
thanks!

**Pull request merged**